### ISSUE TITLE
[Android] Crash on new arch when lottie animation finishes

### ISSUE THREAD

**Issue by alfonsocj** [lottie-react-native/lottie-react-native#963]
### Description
Lottie animation crashes the app when the animation finish event is triggered.
Only reproducible when the RN new arch is enabled.

### Steps to Reproduce

- Open `apps/fabric/App.tsx`
- Set `loop={false}` (or remove `loop`) in `AnimatedLottieView`
- Run `yarn android` in `apps/fabric`
- Wait for the animation to finish.

**Expected behavior:** [What you expected to happen]
The animation finishes without the app crashing

**Actual behavior:** [What actually happened]
App crashes with:
```
E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.airbnb.android.lottie.rnexample, PID: 6274
    java.lang.AssertionError: Method overloading is unsupported: com.facebook.react.uimanager.events.RCTModernEventEmitter#receiveEvent
        at com.facebook.react.bridge.JavaScriptModuleRegistry$JavaScriptModuleInvocationHandler.<init>(JavaScriptModuleRegistry.java:66)
        at com.facebook.react.bridge.JavaScriptModuleRegistry.getJavaScriptModule(JavaScriptModuleRegistry.java:41)
        at com.facebook.react.bridge.CatalystInstanceImpl.getJSModule(CatalystInstanceImpl.java:464)
        at com.facebook.react.bridge.ReactContext.getJSModule(ReactContext.java:160)
        at com.airbnb.android.react.lottie.LottieAnimationViewManager.sendOnAnimationFinishEvent(LottieAnimationViewManager.kt:57)
        at com.airbnb.android.react.lottie.LottieAnimationViewManager.access$sendOnAnimationFinishEvent(LottieAnimationViewManager.kt:31)
        at com.airbnb.android.react.lottie.LottieAnimationViewManager$createViewInstance$1.onAnimationEnd(LottieAnimationViewManager.kt:87)
        at android.animation.Animator$AnimatorListener.onAnimationEnd(Animator.java:600)
        at com.airbnb.lottie.utils.BaseLottieAnimator.notifyEnd(BaseLottieAnimator.java:74)
        at com.airbnb.lottie.utils.LottieValueAnimator.doFrame(LottieValueAnimator.java:103)
        at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1229)
        at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1239)
        at android.view.Choreographer.doCallbacks(Choreographer.java:899)
        at android.view.Choreographer.doFrame(Choreographer.java:827)
        at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1214)
        at android.os.Handler.handleCallback(Handler.java:942)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loopOnce(Looper.java:201)
        at android.os.Looper.loop(Looper.java:288)
        at android.app.ActivityThread.main(ActivityThread.java:7898)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
I/Process: Sending signal. PID: 6274 SIG: 9
```

The crash happens when the `receiveEvent` method gets called in `sendOnAnimationFinishEvent` https://github.com/lottie-react-native/lottie-react-native/blob/master/packages/core/android/src/newarch/com/airbnb/android/react/lottie/LottieAnimationViewManager.kt#L57.

### Versions
Latest master (commit [5bff05c159852ecd06d9a0206dc525c18e8a07f9](https://github.com/lottie-react-native/lottie-react-native/commit/5bff05c159852ecd06d9a0206dc525c18e8a07f9))

**Comment by alfonsocj** [comment]
👋 @matinzd, even though you are aware of it, I'm creating the issue for better visibility.

**Comment by alfonsocj** [comment]
cc @AlexanderEggers

**Comment by UmerQadri** [comment]
@alfonsocj @matinzd any quick workaround?

**Comment by alfonsocj** [comment]
@UmerQadri This won't affect any active version of this library as the new arch support has not been released yet.

**Comment by matinzd** [comment]
I was trying to understand what was going on but no luck for now. Do you have any idea how to solve this? @AlexanderEggers

**Comment by AlexanderEggers** [comment]
@matinzd @alfonsocj I finally got some time and I think I fixed as part of #1000. Can one of you please verify that the changes make sense + verify that the event handling is running as expected?

**Comment by matinzd** [comment]
I can verify that everything is working as expected. Thank you!
